torchtitan/train.py

```diff
@@ -309,7 +309,13 @@ def __init__(self, job_config: JobConfig):
             checkpoint_config=job_config.checkpoint,
             sd_adapter=(
                 self.train_spec.state_dict_adapter(
-                    model_args, job_config.model.hf_assets_path
+                    model_args,
+                    (
+                        job_config.model.hf_assets_path
+                        if job_config.checkpoint.enable
```

Comment:
Could you please move using checkpoint information out of the trainer, if possible? We deliberately hide `job_config.checkpoint` inside Checkpointer and let the trainer always call checkpointer API. This usage looks okay but would be good to hide if possible.